src/journal/journal-file.h

```diff
@@ -85,6 +85,7 @@ typedef struct JournalFile {
         bool compress_lz4:1;
         bool seal:1;
         bool defrag_on_close:1;
+        volatile bool archive:1;
```

Comment:
The compiler is not allowed to cache the value here since you call an external function  : fsync.
Really, unless you do a busy loop without yield (which is very very bad, you only see this kind of code in embedded device), volatile is *never* necessary. And if you do some sharing with other thread : *always* use mutex/spinlock